src/engine.cc

```diff
@@ -210,20 +220,57 @@ void EngineController::SetPosition(const std::string& fen,
 }
 
 void EngineController::Go(const GoParams& params) {
-  if (!tree_) {
-    SetPosition(ChessBoard::kStartingFen, {});
+  go_params_ = params;
+
+  ThinkingInfo::Callback info_callback(info_callback_);
+
+  if(current_position_) {
+    if (params.ponder && !(*current_position_).moves.empty()) {
+      std::vector<std::string> moves((*current_position_).moves);
```

Comment:
Consider implementing `operator->` in `utils/optional.h`
(Which will exactly the same implementation as `operator*`)